pdns/lua-recursor4.cc

```diff
@@ -516,8 +553,10 @@ bool RecursorLua4::genhook(luacall_t& func, const ComboAddress& remote,const Com
   dq->ednsOptions = ednsOpts;
   dq->isTcp = isTcp;
   dq->policyTags = policyTags;
+  dq->appliedPolicy = appliedPolicy;
   bool handled=func(dq);
   if(variable) *variable |= dq->variable; // could still be set to indicate this *name* is variable, even if not 'handled'
+  appliedPolicy = dq->appliedPolicy; // The policy might have been changed, but the query can still be not handled by Lua
```

Comment:
If I'm not mistaken some hooks are calling `genhook()` with `nullptr` in `appliedPolicy`.